common/src/main/java/com/microsoft/identity/common/internal/cache/MsalOAuth2TokenCache.java

```diff
@@ -235,14 +281,58 @@ public ICacheRecord save(@NonNull final GenericOAuth2Strategy oAuth2Strategy,
         result.setAccount(accountToSave);
         result.setAccessToken(accessTokenToSave);
         result.setRefreshToken(refreshTokenToSave);
+        setToCacheRecord(result, idTokenToSave);
 
-        if (CredentialType.V1IdToken.name().equalsIgnoreCase(idTokenToSave.getCredentialType())) {
-            result.setV1IdToken(idTokenToSave);
-        } else {
-            result.setIdToken(idTokenToSave);
+        return result;
+    }
+
+    @Override
+    @NonNull
+    public List<ICacheRecord> saveAndLoadAggregatedAccountData(
+            @NonNull final GenericOAuth2Strategy oAuth2Strategy,
+            @NonNull final GenericAuthorizationRequest request,
+            @NonNull final GenericTokenResponse response) throws ClientException {
+        return mergeCacheRecordWithCorollaryAccounts(
+                save(oAuth2Strategy, request, response)
+        );
+    }
+
+    /**
+     * Given an AccountRecord and associated client_id, load a sparse ICacheRecord containing
+     * the provided AccountRecord and its accompanying IdTokens.
+     *
+     * @param clientId The client_id relative to which IdTokens should be loaded.
+     * @param acct     The target AccountRecord.
+     * @return A sparse ICacheRecord containing the provided AccountRecord and its IdTokens.
+     */
+    ICacheRecord getSparseCacheRecordForAccount(@NonNull final String clientId,
+                                                @NonNull final AccountRecord acct) {
+        final String methodName = ":getSparseCacheRecordForAccount";
+
+        final List<IdTokenRecord> acctIdTokens = getIdTokensForAccount(
+                clientId,
+                acct
+        );
+
+        if (acctIdTokens.size() > new CredentialType[]{
```

Comment:
I think it might be simpler to have a constant here....

Reply:
Resolved in 1e724f6ec32bb12e997da4af53a223817cf15a1f